pkg/module/filter.go

```diff
@@ -189,13 +190,94 @@ func initFromConfig(filePath string) (*Filter, error) {
 	return f, nil
 }
 
+// matches checks if the given version matches the given qualifier.
+// Qualifiers can be:
+// - plain versions
+// - v1.2.3 enables v1.2.3
+// - ~1.2.3: enables 1.2.x  which are at least 1.2.3
+// - ^1.2.3: enables 1.x.x which are at least 1.2.3
+// - <1.2.3: enables everything lower than 1.2.3 includes 1.2.2 and 0.58.9 as well
+func matches(version, qualifier string) bool {
+	prefix := qualifier[0]
+	first := qualifier[1]
+
+	// v1.2.3 means we accept every version starting with v.1.2.3
+	// handle this special case first, then go for ~v1.2.3 and similar
+	if prefix == 'v' && first >= '0' && first <= '9' { // a number
+		return strings.HasPrefix(version, qualifier)
+	}
+
+	v, err := getVersionSegments(version[1:])
+	if err != nil {
```

Comment:
Passing the logger down there would involve messing up with a lot of layers. 
Not sure if it's worth the effort. That function might fail only if an element of the semver is not a string which would not mach any filter by definition.
On the other hand, I was thinking that we could validate the filter file after loading it. It would prevent having invalid qualifiers.

Reply:
Obviuosuly when I say "the effort" I mean the effort in adding that kind of complexity to the codebase, not that I don't want to write the code :-)